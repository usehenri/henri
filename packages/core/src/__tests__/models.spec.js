const BaseModule = require('../base/module');
const Henri = require('../henri');
const Model = require('../2.model');

xdescribe('models', () => {
  beforeAll(async () => {
    this.henri = new Henri({
      runlevel: 4,
    });
    await this.henri.init();
  });

  afterAll(async () => {
    await this.henri.stop();
  });

  test('should be defined', () => {
    expect(this.henri.model).toBeDefined();
  });

  test('should extend BaseModule', () => {
    expect(this.henri.model).toBeInstanceOf(BaseModule);
  });

  test('should match snapshot', () => {
    const model = new Model();
    expect(model).toMatchSnapshot();
  });
});
