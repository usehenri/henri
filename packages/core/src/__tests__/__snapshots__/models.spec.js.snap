// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models should match snapshot 1`] = `
Model {
  "addToEslintRc": [Function],
  "checkStoreOrDie": [Function],
  "configure": [Function],
  "consoleOnly": false,
  "getStore": [Function],
  "henri": null,
  "ids": Array [],
  "init": [Function],
  "key": null,
  "load": [Function],
  "loadStore": [Function],
  "models": Array [],
  "name": "model",
  "reload": [Function],
  "reloadable": true,
  "reset": [Function],
  "runlevel": 4,
  "start": [Function],
  "stop": [Function],
  "stores": Object {},
}
`;
